contrib/babelfishpg_tds/src/backend/tds/tdscomm.c

```diff
@@ -886,13 +930,27 @@ TdsReadNextRequest(StringInfo message, uint8_t *status, uint8_t *messageType)
 		if (TdsGetbytes(message->data + message->len, readBytes))
 			return EOF;
 		message->len += readBytes;
+
+		if (TdsRecvMessageType == TDS_BULK_LOAD)
+		{
+			/* if this is the last packet then notify the caller. */
+			if (TdsRecvPacketStatus & TDS_PACKET_HEADER_STATUS_EOM)
+				BulkCopyPacketStatus = 1;
```

Comment:
this is not the right approach to set a global variable here. Better would to have a new function(or use existing one) to enquire for header status from bulk copy code